woocommerce/woocommerce-gutenberg-products-block
# Add Checkout Order Summary inner block tests

This PR adds tests for the Order Summary inner block.

There was no need to add tests for this block in the editor context due to the only control changing the `showRateAfterTaxName` which should be tested further down the stack in the `TotalsTaxes` component.

I added two utility functions to the `tests/utils` folder which allow you to search for text across elements.

The `textContentMatcher` function will allow you to find text when it is structured like this:

```
<div>Some <span>nested text</span></div>
```

and `textContentMatcherAcrossSiblings` will let you find text when it is structured like this:
```
<div>
    <span>Text</span>
    <span>nested</span>
    <span>in sibling</span>
    <span>elements.</span>
</div>
```

Fixes #5625




#### Accessibility




#### Other Checks


### Screenshots



### Testing

#### Automated Tests
### Manual Testing

How to test the changes in this Pull Request:

1. `npm run test` and ensure the tests pass

There are no user-facing testing instructions required.



### Changelog

> Skipping
